src/mount/mastercomm.cc

```diff
@@ -806,7 +814,19 @@ int fs_connect(bool verbose) {
 			}
 #else
 			fprintf(stderr, " ; root mapped to %" PRIu32 ":%" PRIu32, rootuid, rootgid);
-			fprintf(stderr, " ; users mapped to %" PRIu32 ":%" PRIu32, mapalluid, mapallgid);
+			if (mapalluid != 999 || mapallgid != 999) {
+				if (*mounting_uid_ != USE_LOCAL_ID && *mounting_gid_ != USE_LOCAL_ID) {
+					fprintf(stderr, " ; master server overwrote users mapping");
+				}
+				*mounting_uid_ = mapalluid;
+				*mounting_gid_ = mapallgid;
+			}
+			if (*mounting_uid_ == USE_LOCAL_ID && *mounting_gid_ == USE_LOCAL_ID) {
+				fprintf(stderr, " ; users mapped to local IDs");
+			} else {
+				fprintf(stderr, " ; users mapped to %" PRIu32 ":%" PRIu32,
+				        *mounting_uid_, *mounting_gid_);
+			}
```

Comment:
Anyway an error will be printed?

Reply:
Before these changes, it was showing users mapped to 999:999 when no master side mapping was applied. The 999 are the default values the master sends to the client when no mapping is done at all, this is confusing if take into account we are also probably doing another client side mapping.